lib/checklists/change_note.rb

```diff
@@ -1,23 +1,40 @@
 class Checklists::ChangeNote
+  include ActiveModel::Validations
+
   CONFIG_PATH = Rails.root.join('lib/checklists/change_notes.yaml')
 
+  validates_presence_of :action_id
+  validates_inclusion_of :type, in: %w(addition content_change)
+  validates_format_of :time, with: /\d{4}-\d{2}-\d{2} \d{2}:\d{2}/
+
+  validate do
+    validates_presence_of :note if type == "content_change"
+  end
```

Comment:
I think this block can be
```suggestion
validates_presence_of :note, if: type == "content_change"
```

Reply:
I don’t think that would work, because it needs to lazily execute on the instance, not the class (i.e. `type` will be nil). I think you can do some lambda thing for some of these validations, but I was thinking the `validate` form was simpler.

Reply:
Sort of - updated as it does seem more readable.